uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request-ctrl.js

```diff
@@ -45,6 +45,11 @@ define([
        * @return {Promise}
        */
       vm.calculateToilExpiryDate = function () {
+        //do not calculate expiry date if its already set by manager for this leave request
+        if (vm.isRole('owner') && vm.isMode('edit')) {
+          return $q.resolve(vm.request.toil_expiry_date);
```

Comment:
1. It is to skip the call to `AbsenceType.calculateToilExpiryDate` in case `toil_expiry_date` is already modified by the manager.
2. see above point 
3. yes, TODO is still valid